### PR TITLE
c7n-mailer | Add Slack incoming webhook support

### PR DESCRIPTION
**Pull request by LykinsN**

### CODE REVIEW

File: tools/c7n_mailer/README.md [hunk 1]
```diff
@@ -136,24 +136,26 @@ policies:
...
+
 | Required? | Key                  | Type             | Notes                               |
 |:---------:|:---------------------|:-----------------|:------------------------------------|
 |           | `slack://owners`          | string      | Send to the recipient list generated within email delivery logic |
 |           | `slack://foo@bar.com`     | string      | Send to the recipient specified by email address foo@bar.com |
-|           | `slack://#custodian-test` | string      | Send to the Slack channel indicated in string, i.e. #custodian-test | 
- 
- 
+|           | `slack://#custodian-test` | string      | Send to the Slack channel indicated in string, i.e. #custodian-test |
+|           | `slack://webhook/#c7n-webhook-test` | string      | Send to a Slack webhook; appended with the target channel. |
```
Comment (kapilt) [inline tools/c7n_mailer/README.md]: re notes. its really send to a slack channel via a web hook. the webhook is just an implementation detail of the transport.

File: tools/c7n_mailer/c7n_mailer/utils.py [hunk 2]
```diff
@@ -296,13 +297,13 @@ def kms_decrypt(config, logger, session, encrypted_field):
                     'Plaintext']
         except (TypeError, base64.binascii.Error) as e:
             logger.warning(
-                "Error: %s Unable to base64 decode %s, will assume plaintext.",
+                "Error: %s Unable to base64 decode %s, will assume plaintext." %
```
Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/utils.py]: py std logging will lazy  render these if passed in as method arguments based on configured logging level.

File: tools/c7n_mailer/c7n_mailer/utils.py [hunk 3]
```diff
@@ -296,13 +297,13 @@ def kms_decrypt(config, logger, session, encrypted_field):
...
                 (e, encrypted_field))
         except ClientError as e:
             if e.response['Error']['Code'] != 'InvalidCiphertextException':
                 raise
             logger.warning(
-                "Error: %s Unable to decrypt %s with kms, will assume plaintext.",
+                "Error: %s Unable to decrypt %s with kms, will assume plaintext." %
```
Comment (kapilt) [inline tools/c7n_mailer/c7n_mailer/utils.py]: same here re lazy render if pass as args to log method

**Comment by LykinsN** [comment]
Rolling these changes into https://github.com/capitalone/cloud-custodian/pull/2347 and closing.

**Pull request closed without merging**